File: Selenium/Activities/Activity16.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService

serv = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=serv) as driver:
    wait = WebDriverWait(driver, 10)
    driver.get("https://www.training-support.net/selenium/dynamic-attributes")
    print("Page title is: ", driver.title)
    username = driver.find_element(By.XPATH, "//input[contains(@class, '-username')]")
    password = driver.find_element(By.XPATH, "//input[contains(@class, '-password')]")
    confirm_password = driver.find_element(By.XPATH, "//label[contains(text(), 'Confirm Password')]//following::input")
    email = driver.find_element(By.XPATH, "//input[contains(@class, 'email-')]")

    username.send_keys("admin")
    password.send_keys("password")
    confirm_password.send_keys("password")
    email.send_keys("adm.in@sample.com")
    driver.find_element(By.XPATH, "//button[text()='Sign Up']").click()
    wait.until(expected_conditions.visibility_of_element_located((By.ID, "action-confirmation")))
    message = driver.find_element(By.ID, "action-confirmation").text
    print("Success message: ", message)

File: Selenium/Activities/Activity16.java
package Activities;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedCondition;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity16 {
    public static void main(String[] args) throws InterruptedException {
        WebDriverManager.firefoxdriver().setup();
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriver newDriver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(newDriver,Duration.ofSeconds(10));
        newDriver.get("https://training-support.net/selenium/dynamic-attributes");
        System.out.println("Title is: " + newDriver.getTitle());
        WebElement usernameEle = newDriver.findElement(By.cssSelector("input[class$='-username']"));
        usernameEle.sendKeys("sample");
        WebElement pwdEle = newDriver.findElement(By.cssSelector("input[class$='-password']"));
        pwdEle.sendKeys("test1234");
        newDriver.findElement(By.xpath("//label[text()='Confirm Password']/following-sibling::input")).sendKeys("test1234");
        newDriver.findElement(By.xpath("//input[@type='email']")).sendKeys("abc@example.com");
        newDriver.findElement(By.xpath("//button[contains(text(),'Sign Up')]")).click();
        wait.until(ExpectedConditions.presenceOfElementLocated(By.id("action-confirmation")));
        System.out.println("Message is: " + newDriver.findElement(By.id("action-confirmation")).getText());
        Thread.sleep(300);
        newDriver.quit();
    }
}
